packages/shorebird_cli/test/src/commands/init_command_test.dart

```diff
@@ -134,24 +138,64 @@ Please make sure you are running "shorebird init" from the root of your Flutter
       expect(exitCode, ExitCode.software.code);
     });
 
-    test('detects existing shorebird.yaml', () async {
-      const existingAppId = 'existing-app-id';
+    test('throws software error when unable to fetch apps.', () async {
+      const error = 'oops something went wrong';
+      final tempDir = Directory.systemTemp.createTempSync();
+      when(() => codePushClient.getApps()).thenThrow(error);
+      File(
+        p.join(tempDir.path, 'pubspec.yaml'),
+      ).writeAsStringSync(pubspecYamlContent);
+      File(
+        p.join(tempDir.path, 'shorebird.yaml'),
+      ).writeAsStringSync('app_id: $appId');
+      final exitCode = await IOOverrides.runZoned(
+        command.run,
+        getCurrentDirectory: () => tempDir,
+      );
+      verify(() => progress.fail(error)).called(1);
+      expect(exitCode, ExitCode.software.code);
+    });
+
+    test('detects existing shorebird.yaml (existing app)', () async {
```

Comment:
Not blocking, but putting this test and the following in a `group('existing Shorebird app')` might make the test names a bit cleaner.

Reply:
I adjusted the test names to be a bit clearer lmk what you think. One of the two tests is for an existing shorebird app and the other is for a non-existing shorebird app.